radis/lbl/loader.py

```diff
@@ -1323,21 +1336,49 @@ def fetch_databank(
                 )
             if len(frames) > 1:
                 # Note @dev : may be faster/less memory hungry to keep lines separated for each isotope. TODO : test both versions
-                for df in frames:
-                    if "iso" not in df.columns:
-                        assert "iso" in df.attrs
-                        df["iso"] = df.attrs["iso"]
-                # Keep attributes:
-                from radis.misc.basics import intersect
-
-                attrs = frames[0].attrs
-                for df in frames[1:]:
-                    attrs = intersect(attrs, df.attrs)
-                del attrs["iso"]  # added as a column (different for each line)
-                # Merge:
-                df = pd.concat(frames, ignore_index=True)  # reindex
-                df.attrs = attrs
-                self.params.dbpath = ",".join(local_paths)
+                if output == "pandas":
+                    for df in frames:
+                        if "iso" not in df.columns:
+                            assert "iso" in df.attrs
+                            df["iso"] = df.attrs["iso"]
+                    # Keep attributes:
+                    from radis.misc.basics import intersect
+
+                    attrs = frames[0].attrs
+                    for df in frames[1:]:
+                        attrs = intersect(attrs, df.attrs)
+                    del attrs["iso"]  # added as a column (different for each line)
+                    # Merge:
+                    df = pd.concat(frames, ignore_index=True)  # reindex
+                    df.attrs = attrs
+                    self.params.dbpath = ",".join(local_paths)
+                elif output == "vaex":
+                    Frames = []
+                    for df in frames:
+                        if "iso" not in df.columns:
+                            assert "iso" in df.attrs
+                            import vaex
+
+                            df["iso"] = vaex.vconstant(
+                                int(df.attrs["iso"]), length=df.length_unfiltered()
+                            )
+                            Frames.append(df)
+                    # Keep attributes:
+                    from radis.misc.basics import intersect
+
+                    attrs = Frames[0].attrs
+                    for df in Frames[1:]:
+                        attrs = intersect(attrs, df.attrs)
+                    del attrs["iso"]  # added as a column (different for each line)
+                    # Merge:
+                    import vaex
+
+                    df = vaex.concat(Frames)  # reindex
+                    df.attrs = attrs
+
+                    self.params.dbpath = ",".join(local_paths)
```

Comment:
@1someshverma @minouHub ; the "iso" as an attribute rather than a column is just a Pandas trick to save memory... Given than Vaex doesn't use any memory at all, I don't think it is necessary in the Vaex case

Reply:
I think that it will cost more time if we store "iso" as a column instead of an attribute.

Reply:
sure about that ?